K2Bridge/Visitors/RangeVisitor.cs

```diff
@@ -0,0 +1,66 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+// See LICENSE file in the project root for full license information.
+
+namespace K2Bridge.Visitors
+{
+    using System.Globalization;
+    using K2Bridge.Models.Request.Aggregations;
+    using K2Bridge.Models.Response;
+
+    /// <content>
+    /// A visitor for the <see cref="RangeAggregation"/> element.
+    /// </content>
+    internal partial class ElasticSearchDSLVisitor : IVisitor
+    {
+        /// <inheritdoc/>
+        public void Visit(RangeAggregation rangeAggregation)
+        {
+            Ensure.IsNotNull(rangeAggregation, nameof(TermsAggregation));
+            EnsureClause.StringIsNotNullOrEmpty(rangeAggregation.Metric, nameof(RangeAggregation.Metric));
+            EnsureClause.StringIsNotNullOrEmpty(rangeAggregation.Field, nameof(RangeAggregation.Field));
+
+            // Start the union operator
+            rangeAggregation.KustoQL += "union ";
+
+            // Query expressions for each range
+            // (_data | where foo >= 1 and bar < 10 | summarize ['3']=avg(baz), count() | extend ['2'] = '1-10')
+            foreach (var range in rangeAggregation.Ranges)
+            {
+                rangeAggregation.KustoQL += $"(_data | {KustoQLOperators.Where} ";
+
+                var rangeClause = new Models.Request.Queries.RangeClause()
+                {
+                    FieldName = rangeAggregation.Field,
+                    GTEValue = range.From?.ToString() ?? "*",
+                    LTValue = range.To?.ToString() ?? "*",
+                };
+                rangeClause.Accept(this);
+
+                if (string.IsNullOrEmpty(rangeClause.KustoQL))
+                {
+                    // This is then "open" range, -infinity to +infinity
+                    rangeAggregation.KustoQL += "true";
+                }
+                else
+                {
+                    rangeAggregation.KustoQL += $"{rangeClause.KustoQL}";
```

Comment:
Is this not equivalent to `rangeAggregation.KustoQL += rangeClause.KustoQL;` ?

Reply:
Absolutely; copy/paste gone wrong :) Fixed.